### ISSUE TITLE
Exception after reseting input from the server

### ISSUE THREAD

**Issue by rxcolvin** [kwebio/kweb-core#420]
Hi,

At certain point after updating the kvar value on the server side  I get the exception below when entering data into the field on the client.

Using the sample code below it can be reproduced by:
1. clicking "reset"
2. Entering "12345" into the field
3. Entering 1 into the field
```
[qtp508198356-55] ERROR kweb.html.events.OnReceiver - Exception thrown by callback in response to input event
java.lang.StringIndexOutOfBoundsException: begin 0, end 1, length 0
	at java.base/java.lang.String.checkBoundsBeginEnd(String.java:4601)
	at java.base/java.lang.String.substring(String.java:2704)
	at kweb.ValueElement.applyDiff(prelude.kt:641)
	at kweb.ValueElement.access$applyDiff(prelude.kt:555)
	at kweb.ValueElement$updateKVar$$inlined$event$1$1.invoke(OnReceiver.kt:190)
	at kweb.ValueElement$updateKVar$$inlined$event$1$1.invoke(OnReceiver.kt:41)
	at kweb.WebBrowser.catchOutbound(WebBrowser.kt:90)
	at kweb.WebBrowser.batch(WebBrowser.kt:97)
	at kweb.ValueElement$updateKVar$$inlined$event$1.invoke(OnReceiver.kt:41)
	at kweb.ValueElement$updateKVar$$inlined$event$1.invoke(OnReceiver.kt:30)
	at kweb.html.events.OnReceiver$event$1.invoke(OnReceiver.kt:21)
	at kweb.html.events.OnReceiver$event$1.invoke(OnReceiver.kt:20)
	at kweb.Element$addEventListener$1.invoke(Element.kt:521)
	at kweb.Element$addEventListener$1.invoke(Element.kt:519)
	at kweb.Kweb.listenForWebsocketConnection(Kweb.kt:302)
	at kweb.Kweb.access$listenForWebsocketConnection(Kweb.kt:42)
	at kweb.Kweb$listenForWebsocketConnection$1.invokeSuspend(Kweb.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034)
	at java.base/java.lang.Thread.run(Thread.java:833)
[qtp508198356-29] ERROR Application - Websocket handler failed
java.lang.IllegalStateException: Unable to find server state corresponding to client id oomMTK:LvLNKn
	at kweb.Kweb.ensureSessionExists(Kweb.kt:255)
	at kweb.Kweb.listenForWebsocketConnection(Kweb.kt:267)
	at kweb.Kweb.access$listenForWebsocketConnection(Kweb.kt:42)
	at kweb.Kweb$listenForWebsocketConnection$1.invokeSuspend(Kweb.kt)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:883)
	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:1034)
	at java.base/java.lang.Thread.run(Thread.java:833)

```
Sample Code
```
import kweb.*
import kweb.components.Component
import kweb.state.KVar

fun main() {
    Kweb(
        port = 2395,
        plugins = listOf(
        )
    ) {
        doc.body {
            val state1 = kvar("foo")
            div().new {
                textEditor(state1)
            }

            div().new {
                with(button()) {
                    text("reset")
                    on.click {
                        state1.value = ""
                    }
                }
            }

            div().new {
                with(button()) {
                    text("print")
                    on.click {
                        println("value1 =${state1.value}")
                    }
                }

            }
        }
    }
}


fun Component.textEditor(
    state: KVar<String>,
) {
    val inputElement = input()
    inputElement.value = state
}

```

**Comment by sanity** [comment]
This likely relates to the previous change, I'll investigate.

**Comment by sanity** [comment]
Having trouble reproducing:

> 1. clicking "reset"
> 2. Entering "12345" into the field
> 3. Entering 1 into the field

Are you hitting reset again between steps 2 and 3?

**Comment by sanity** [comment]
Actually, never mind - looks like reset between 2 and 3 does it.

**Comment by sanity** [comment]
Fixed, will deploy in a new version soon.

**Comment by sanity** [comment]
Fixed in `1.3.4`.

**Comment by rxcolvin** [comment]
great, thx. apologies for the slight mis-direction.
